sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3749,9 +3713,11 @@ slm_gather(simd<uint32_t, N / VS> byte_offsets, simd_mask<N / VS> mask,
   static_assert(Alignment >= sizeof(T),
                 "slm_gather() requires at least element-size alignment");
 
-  if constexpr (VS > 1) {
+  if constexpr (VS > 1 || (!detail::isPowerOf2(N, 32) &&
+                           !detail::isMaskedGatherScatterLLVMAvailable())) {
+    simd<T, N> pass_thru;
```

Comment:
nit: please add the comment that it is intentionally undefined. Also, per clang coding std, the var should be PassThru.